### PR TITLE
player-mpris-tail: Refactor and fix metadata parsing function

### PR DESCRIPTION
**Pull request by rockyprabowo**
This PR refactors `Player._parseMetadata` to increase the readability and also adds some flexibility to the function in some cases that might occur in the future like strange or non-conformant metadata format.
In my case, I've found that cmus sends `xesam:title` and `xesam:album` as an array of string (even though it only contains one string) instead of a string as per MPRIS metadata specification, causing this error to occur:
```
Traceback (most recent call last):
  File "/home/rocky-i3/Scripts/polybar-player-mpris-tail", line 509, in <module>
    PlayerManager(blacklist = args.blacklist)
  File "/home/rocky-i3/Scripts/polybar-player-mpris-tail", line 32, in __init__
    self.refreshPlayerList()
  File "/home/rocky-i3/Scripts/polybar-player-mpris-tail", line 84, in refreshPlayerList
    self.addPlayer(player_bus_name)
  File "/home/rocky-i3/Scripts/polybar-player-mpris-tail", line 90, in addPlayer
    player = Player(self._session_bus, bus_name, owner = owner, connect = self._connect, _print = self.print)
  File "/home/rocky-i3/Scripts/polybar-player-mpris-tail", line 196, in __init__
    self.refreshMetadata()
  File "/home/rocky-i3/Scripts/polybar-player-mpris-tail", line 246, in refreshMetadata
    self._parseMetadata()
  File "/home/rocky-i3/Scripts/polybar-player-mpris-tail", line 280, in _parseMetadata
    self.metadata['album']  = re.sub(SAFE_TAG_REGEX, """\1\1""", _getProperty(self._metadata, 'xesam:album', ''))
  File "/usr/lib/python3.8/re.py", line 208, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: expected string or bytes-like object
```

### CODE REVIEW

**Comment by x70b1** [comment]
@Cybolic

File: polybar-scripts/player-mpris-tail/player-mpris-tail.py [hunk 1]
```diff
@@ -264,45 +264,34 @@ def _print(self, status):
...
-            self.metadata['duration'] = _getDuration(self.metadata['length'])
+            # Obtain properties from _metadata
+            _artist     = _getProperty(self._metadata, 'xesam:artist', [''])
+            _album      = _getProperty(self._metadata, 'xesam:album', '')
+            _title      = _getProperty(self._metadata, 'xesam:title', '')
+            _track      = _getProperty(self._metadata, 'xesam:trackNumber', '')
+            _genre      = _getProperty(self._metadata, 'xesam:genre', [''])
+            _disc       = _getProperty(self._metadata, 'xesam:discNumber', '')
+            _length     = _getProperty(self._metadata, 'xesam:length', 0) or _getProperty(self._metadata, 'mpris:length', 0)
+            _length_int = _length if type(_length) is int else int(_length)
```
Comment (Cybolic) [inline polybar-scripts/player-mpris-tail/player-mpris-tail.py]: I think we still need `int(float(_length))` here instead of just `int(_length)`.

**Comment by Cybolic** [comment]
Apart from that one comment, this looks great. Much cleaner! :D

File: polybar-scripts/player-mpris-tail/player-mpris-tail.py (hunk 1 above)
Comment (rockyprabowo) [reply polybar-scripts/player-mpris-tail/player-mpris-tail.py]: @Cybolic, I've force-pushed the small change per your request.
 If I have to change something let me know.

**Comment by Cybolic** [comment]
I have nothing to add, it's great!

**Comment by x70b1** [comment]
Thx! @rockyprabowo @Cybolic

**Pull request merged**